sips/sip-012/sip-012-cost-limits-network-upgrade.md

```diff
@@ -0,0 +1,942 @@
+memory usage of `var-set` remains in effect until the end of the
+transaction (data operations remain in memory during the whole
+transaction to enable rollbacks and post-conditions).
+
+#### `(map-get? map-name key) -> value`
+
+The `x` input to the `map-get` cost function should be the sum of the
+length in bytes of the consensus serialization of the supplied `key`
+and the returned `value`.
+
+#### `(map-set map-name key value)`
+
+The `x` input to the `map-set` cost function should be the sum of the
+length in bytes of the consensus serialization of the supplied `key` and
+`value` arguments.
+
+The memory usage of this function should be this same value. The
+memory usage of `map-set` remains in effect until the end of the
+transaction (data operations remain in memory during the whole
+transaction to enable rollbacks and post-conditions).
+
+#### `(map-insert map-name key value)`
+
+If the insert is successful, the `x` input to the `map-insert` cost
+function should be the sum of the length in bytes of the consensus
+serialization of the supplied `key` and `value` arguments.
+
+If the insert is unsuccessful, the `x` input to the `map-insert` cost
+function should be the length in bytes of the consensus serialization
+of just the supplied `key` argument.
+
+The memory usage of this function should be this same `x` value. The
+memory usage of `map-insert` remains in effect until the end of the
+transaction (data operations remain in memory during the whole
+transaction to enable rollbacks and post-conditions).
+
+#### `(map-delete map-name key)`
+
+The `x` input to the `map-delete` cost function should be the length
+in bytes of the consensus serialization of the supplied `key`
+argument plus the length in bytes of the consensus serialization of
+a `none` Clarity value.
+
+The memory usage of this function should be this same `x` value. The
+memory usage of `map-delete` remains in effect until the end of the
+transaction (data operations remain in memory during the whole
+transaction to enable rollbacks and post-conditions).
+
+#### `(concat list-a list-b)`
+
+The `x` input to the `concat` cost function should be the length of
+`list-a` plus the length of `list-b`.
+
+### New Default Cost Functions
+
+Based on results from the
+[clarity-benchmarking](https://github.com/blockstack/clarity-benchmarking)
+project, this SIP proposes new default cost functions. The new costs are supplied in
+the form of a new Clarity smart contract in [Appendix A](#appendix-a).
+
+This could have been carried out through a SIP-006 cost voting procedure, but
+due to the opportunity costs incurred by STX holders foregoing PoX rewards to
+carry this vote out, this SIP instead proposes bypassing the SIP-006 voting
+procedure in this one instance and instead using this SIP's activation procedure
+to install these new functions.
+
+# Activation
+
+The SIP will be considered Ratified once all of the following are true:
```

Comment:
Add another criterion: the cost functions have to be finalized.  Thanks @friedger!